n3fit/src/n3fit/layers/msr_normalization.py

```diff
@@ -51,30 +83,35 @@ def call(self, pdf_integrated, photon_integral):
 
         Parameters
         ----------
-        pdf_integrated: (Tensor(1,None,14))
+        pdf_integrated: (Tensor(1, 14))
             the integrated PDF
-        photon_integral: (Tensor(1)):
-            the integrated photon, not included in PDF
+        photon_integral: (Tensor(1, 1))
+            the integrated photon PDF
 
         Returns
         -------
         normalization_factor: Tensor(14)
             The normalization factors per flavour.
         """
-        y = op.flatten(pdf_integrated)
+        y = pdf_integrated[0]  # get rid of the batch dimension
+        photon_integral = photon_integral[0]  # get rid of the batch dimension
         norm_constants = []
 
         if self._msr_enabled:
-            n_ag = [(1.0 - y[GLUON_IDX[0][0] - 1] - photon_integral[0]) / y[GLUON_IDX[0][0]]] * len(
-                GLUON_IDX
-            )
-            norm_constants += n_ag
+            norm_constants += [
+                op.batchit(1.0 - y[IDX['sigma']] - photon_integral[0], batch_dimension=0)
+            ]
```

Comment:
Why do you need to "batch this" while it wasn't necessary before?

Reply:
This is because before the vsr factors were a list of scalars, which I've turned into a rank 1 tensor. So to be able to concatenate that with the msr factor, that needs to be a rank 1 tensor too.